pkg/agent/billing/billing.go

```diff
@@ -81,23 +87,40 @@ type vmMetricsSeconds struct {
 }
 
 func RunBillingMetricsCollector(
-	backgroundCtx context.Context,
+	ctx context.Context,
```

Comment:
Curious about this rename- why `backgroundCtx` -> `ctx` ?

I see that this commit (13c5934add9d24cc72d9615b722ebb3d7d4925dc) changes `billing.NewS3Client` to use the context, but IIUC the semantics associated with the context haven't changed

Reply:
It was `backgroundCtx` because it's only used so that the billing subsystem can be notified when the autoscaler-agent is going to exit soon (so we can send any events that are waiting around).

Its direct usage is in the collector thread:

https://github.com/neondatabase/autoscaling/blob/3a2a87ce3609448023cb2421136fb84bcf67ec9a/pkg/agent/billing/billing.go#L187-L188

which in turn causes `(eventSender).senderLoop` to finish up:

https://github.com/neondatabase/autoscaling/blob/3a2a87ce3609448023cb2421136fb84bcf67ec9a/pkg/agent/billing/send.go#L60-L62

Looking at it now though, `senderLoop` should probably just take a context...